singularity-eos/base/constants.hpp

```diff
@@ -32,11 +32,21 @@ constexpr unsigned long all_values = (1 << 7) - 1;
 } // namespace thermalqs
 
 constexpr size_t MAX_NUM_LAMBDAS = 3;
-enum class DataStatus { Deallocated = 0, OnDevice = 1, OnHost = 2 };
+enum class DataStatus { Deallocated = 0, OnDevice = 1, OnHost = 2, UnManaged = 3 };
```

Comment:
Do we have sufficient coverage in our testing to make sure we are catching the potential issues? I recall this took up many of @Yurlungur's cycles to get this to work correctly back in the day.

Reply:
I *believe* so. I think our tests should catch this if there's an issue. But I can't promise that. The `UnManaged` tag just makes `Finalize` a no-op.